onedal/linear_model/incremental_linear_model.py

```diff
@@ -113,36 +114,36 @@ def finalize_fit(self, queue=None):
         Parameters
         ----------
         queue : dpctl.SyclQueue
-            Not used here, added for API conformance
+            If not None, use this queue for computations.
 
         Returns
         -------
         self : object
             Returns the instance itself.
         """
+
+        if queue is not None:
+            policy = self._get_policy(queue)
+        else:
+            policy = self._get_policy(self._queue)
+
         module = self._get_backend("linear_model", "regression")
         hparams = get_hyperparameters("linear_regression", "train")
         if hparams is not None and not hparams.is_default:
             result = module.finalize_train(
-                self._policy, self._params, hparams.backend, self._partial_result
+                policy, self._params, hparams.backend, self._partial_result
             )
         else:
-            result = module.finalize_train(
-                self._policy, self._params, self._partial_result
-            )
+            result = module.finalize_train(policy, self._params, self._partial_result)
 
         self._onedal_model = result.model
 
         packed_coefficients = from_table(result.model.packed_coefficients)
         self.coef_, self.intercept_ = (
-            packed_coefficients[:, 1:],
-            packed_coefficients[:, 0],
+            packed_coefficients[:, 1:].squeeze(),
```

Comment:
Why would squeezing be needed here?

Reply:
Its a change related to this change, ideally IncrementalLinearRegression would be changed to follow changes that were requested in IncrementalRidge, but it looks like the tests were changed instead. https://github.com/intel/scikit-learn-intelex/pull/1972#discussion_r1741984101

Reply:
Thanks to @olegkkruglov my previous comment is very wrong. He has implemented the change that I mention in the step above.  As far as I understand, its because linear regression supports multi_output.  In the case that y is 1 dimensional, the onedal object may return a 2 dimensional object where the size in one of the dimensions is 1 and it must be squeezed out to match the original input y data.

Reply:
But why would squeeze be needed in a test?

Reply:
it was recognized after implementing this change that tests were not exactly correct, and after comparing with original sklearn, i've changed tests as well